File: OpenVehicleApp/src/main/java/com/openvehicles/OVMS/api/ApiObserver.java
package com.openvehicles.OVMS.api;

import com.openvehicles.OVMS.entities.CarData;

public interface ApiObserver {
    void update(CarData pCarData);
    void onServiceAvailable(ApiService pService);
    void onServiceLoggedIn(ApiService pService, boolean pIsLoggedIn);
}

File: app/src/main/java/com/openvehicles/OVMS/api/ApiObserver.kt
package com.openvehicles.OVMS.api

import com.openvehicles.OVMS.entities.CarData

interface ApiObserver {

    fun update(carData: CarData?)

    fun onServiceAvailable(service: ApiService)

    fun onServiceLoggedIn(service: ApiService?, isLoggedIn: Boolean)

}